src/modules/mahjong/PointValidator/PointValidator.ts

```diff
@@ -0,0 +1,338 @@
+import {
+  ValidPair,
+  HandStructureResults,
+  HandDefinition,
+  HandPointResults,
+  PointValidationResults,
+} from '../types/MahjongTypes';
+import MeldEnums from '../enums/MeldEnums';
+import TileMapper from '../Tile/map/TileMapper';
+import HKHandMapper from '../Hand/map/HKHandMapper';
+import {
+  isSimpleTile as isSimpleTileUtils,
+  isHonorTile as isHonorTileUtils,
+  isBonusTile as isBonusTileUtils,
+} from '../utils/functions/checkTypes';
+
+import SimpleTileTypes from '../enums/SimpleTileEnums';
+
+class PointValidator {
+  static MAX_POINTS = 13;
+
+  static MIN_POINTS = 3;
+  // Validate each hand; assume that each validPair has been validated by the HandValidator as
+  // a valid hand and contains an array of melds, otherwise assume the argument is invalid
+
+  /**
+   * Verfies whether a hand contains all consecutives
+   * @returns the corresponding score if valid, otherwise 0
+   */
+  public static validateAllConsecutives = (vp: ValidPair): HandDefinition => {
+    const { melds } = vp;
+    let allConsecutives = true;
+    if (melds) {
+      melds.forEach((meld) => {
+        if (meld.type !== MeldEnums.CONSECUTIVE) allConsecutives = false;
+      });
+
+      if (allConsecutives) return HKHandMapper.ALL_CONSECUTIVE;
+    }
+
+    return HKHandMapper.INVALID;
+  };
+
+  /**
+   * Verfies whether a hand contains all triplets
+   * @returns the corresponding score if valid, otherwise 0
+   */
+  public static validateAllTriplets = (vp: ValidPair): HandDefinition => {
+    const { melds } = vp;
+    let allTriplets = true;
+    if (melds) {
+      melds.forEach((meld) => {
+        if (meld.type !== MeldEnums.TRIPLET && meld.type !== MeldEnums.QUAD) allTriplets = false;
```

Comment:
should the condition be `OR`? i am not sure though haha
```js
if(meld.type !== MeldEnums.TRIPLET || meld.type !== MeldEnums.QUAD) allTriplets = false;
```

Reply:
It should be &&. We're checking that the meld type isn't a triplet and it's also not a quad. If it was ||, then if the meld type was a Triplet, it would still pass because it's not considered a quad.

Reply:
got it